src/sage/categories/rings.py

```diff
@@ -423,6 +423,28 @@ def is_integral_domain(self, proof=True) -> bool:
 
             return False
 
+        def is_integrally_closed(self) -> bool:
+            r"""
+            Return whether this ring is integrally closed.
+
+            This is the default implementation that returns ``False``.
+
+            EXAMPLES::
+
+                sage: x = polygen(ZZ, 'x')
+                sage: K.<a> = NumberField(x^2 + 189*x + 394)
+                sage: R = K.order(2*a)
+                sage: R.is_integrally_closed()
+                False
+                sage: R
+                Order of conductor 2 generated by 2*a in Number Field in a with defining polynomial x^2 + 189*x + 394
+                sage: S = K.maximal_order(); S
+                Maximal Order generated by a in Number Field in a with defining polynomial x^2 + 189*x + 394
+                sage: S.is_integrally_closed()
+                True
+            """
+            return False
```

Comment:
I'm not too familiar with the Sage category setup, but why isn't the default implementation to raise a `NotImplementedError`?

Reply:
yes, indeed, that's another possibility and that was the previous behaviour. We can also decide to say "False" if there is no specific reason known for the property in question to hold. I have no strong opinion.

Reply:
> yes, indeed, that's another possibility and that was the previous behaviour. We can also decide to say "False" if there is no specific reason known for the property in question to hold. I have no strong opinion.

I feel that it's better to raise an error than report a potentially incorrect result.